### PR TITLE
Update Jazzy theme

### PR DESCRIPTION
**Pull request by captainbarbosa**
Edits Jazzy generation script to support https://github.com/realm/jazzy/pull/1077.

Requires https://github.com/mapbox/jazzy-theme/pull/9 to merge before merging.

cc @1ec5

### CODE REVIEW

**Comment by captainbarbosa** [comment]
Running `scripts/document.sh` results in `${RELEASE_VERSION}` being `0.17.0`, which is slightly unexpected to me given that our release page says we're on `0.35.0`:

Screenshot 2019-07-15 13 44 34

**Comment by 1ec5** [comment]
> Running `scripts/document.sh` results in `${RELEASE_VERSION}` being `0.17.0`, which is slightly unexpected to me given that our release page says we're on `0.35.0`:

You may need to run `git fetch --tags` to make sure your clone contains all the tags used to calculate the latest release version.

File: scripts/document.sh [hunk 1]
```diff
@@ -60,10 +60,10 @@ jazzy \
     --documentation="docs/guides/*.md" \
     --root-url "${BASE_URL}/navigation/${RELEASE_VERSION}/" \
     --theme ${THEME} \
-    --output ${OUTPUT}
+    --output ${OUTPUT} \
+    --title "Navigation SDK for iOS" \
```
Comment (1ec5) [inline scripts/document.sh]: Can we move this configuration setting to jazzy.yml?

Comment (captainbarbosa) [reply scripts/document.sh]: Yes, but then I think we won't be able to access `${RELEASE_VERSION}`...?

Comment (1ec5) [reply scripts/document.sh]: Just move `title` to jazzy.yml, but keep `module_version` here.

**Comment by 1ec5** [comment]
> Requires mapbox/jazzy-theme#9 to merge before merging.

I missed this bit. This SDK doesn’t depend on jazzy-theme yet; it hosts its own copy of the theme. We’ll need to make the change here:

https://github.com/mapbox/mapbox-navigation-ios/blob/4b22a6860597131cd7b7fb184fbc818750dd2939/docs/theme/templates/header.mustache#L7

**Pull request merged**